vortex-layout/src/layouts/flat/writer.rs

```diff
@@ -79,7 +71,68 @@ impl LayoutWriter for FlatLayoutWriter {
             vortex_bail!("FlatLayoutStrategy::push_batch called after finish");
         }
         let row_count = chunk.len() as u64;
-        update_stats(&chunk, &self.options.array_stats)?;
+
+        match chunk.dtype() {
+            DType::Utf8(_) => {
+                if let Some(sv) = chunk.statistics().get(Stat::Min) {
+                    let (value, truncated) = lower_bound::<Utf8Scalar>(
+                        chunk.dtype(),
+                        sv.into_inner(),
+                        self.options.max_variable_length_statistics_size,
+                    )?;
+                    if truncated {
+                        chunk.statistics().set(Stat::Min, Precision::Inexact(value));
+                    }
+                }
+
+                if let Some(sv) = chunk.statistics().get(Stat::Max) {
+                    let (value, truncated) = upper_bound::<Utf8Scalar>(
+                        chunk.dtype(),
+                        sv.into_inner(),
+                        self.options.max_variable_length_statistics_size,
+                    )?;
+                    if let Some(upper_bound) = value {
```

Comment:
I don't understand why this logic looks different from the Min / lower bound logic?